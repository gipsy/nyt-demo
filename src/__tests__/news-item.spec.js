import React from 'react'
import { mount } from 'enzyme'
import { initialData } from './config/initial-data'

import NewsItem from '@components/news-item'

describe('NewsItem', () => {
  it('renders without crashing', () => {
    const wrap = mount(<NewsItem item={initialData.news.data[0]} />)
    expect(wrap).toMatchSnapshot()
    wrap.unmount()
  })
})
