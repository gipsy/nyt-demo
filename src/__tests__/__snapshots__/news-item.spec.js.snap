// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsItem renders without crashing 1`] = `
<NewsItem
  item={
    Object {
      "_id": "5ba46d04068401528a2dc251",
      "headline": Object {
        "main": "Singapore, for Those Who Aren’t Crazy Rich",
        "print_headline": "Singapore, if You’re Not Crazy Rich",
      },
      "multimedia": Array [
        Object {
          "subtype": "tumbnail",
          "url": "images/2018/09/23/travel/23singapore1/merlin_142944708_9276e9dd-b36e-40f3-90f6-34e79e64552e-articleLarge.jpg",
        },
      ],
      "pub_date": "2018-09-21T04:01:06+0000",
      "snippet": "In the hit movie “Crazy Rich Asians,” the country looks luxe. But it’s easy to have a great time (and eat memorably well) without spending much.",
      "source": "The New York Times",
      "web_url": "https://www.nytimes.com/2018/09/21/travel/singapore-travel-budget.html",
    }
  }
>
  <ForwardRef
    extra={
      <img
        alt="logo"
        src="https://www.randjsc.com/wp-content/uploads/2015/05/NYT-logo-square-150x150.jpg"
        width={75}
      />
    }
    key="5ba46d04068401528a2dc251"
  >
    <StyledComponent
      extra={
        <img
          alt="logo"
          src="https://www.randjsc.com/wp-content/uploads/2015/05/NYT-logo-square-150x150.jpg"
          width={75}
        />
      }
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "Meta": [Function],
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "styled__Item-npycut-0",
            "isStatic": true,
            "lastClassName": "bCWxja",
            "rules": Array [
              "display:block;",
            ],
          },
          "displayName": "styled__Item",
          "render": [Function],
          "styledComponentId": "styled__Item-npycut-0",
          "target": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        }
      }
      forwardedRef={null}
    >
      <Item
        className="styled__Item-npycut-0 bCWxja"
        extra={
          <img
            alt="logo"
            src="https://www.randjsc.com/wp-content/uploads/2015/05/NYT-logo-square-150x150.jpg"
            width={75}
          />
        }
      >
        <div
          className="ant-list-item styled__Item-npycut-0 bCWxja"
        >
          <div
            className="ant-list-item-extra-wrap"
          >
            <div
              className="ant-list-item-main"
            >
              <div
                className="ant-list-item-content ant-list-item-content-single"
              >
                <ForwardRef>
                  <StyledComponent
                    forwardedClass={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": undefined,
                        "componentStyle": ComponentStyle {
                          "componentId": "styled__Source-npycut-4",
                          "isStatic": true,
                          "lastClassName": "edhqYd",
                          "rules": Array [
                            "font-family:'Libre Franklin',sans-serif;text-transform:uppercase;font-size:10px;font-weight:700;color:rgb(18,18,18);display:block;",
                          ],
                        },
                        "displayName": "styled__Source",
                        "render": [Function],
                        "styledComponentId": "styled__Source-npycut-4",
                        "target": "p",
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <p
                      className="styled__Source-npycut-4 edhqYd"
                    >
                      The New York Times
                    </p>
                  </StyledComponent>
                </ForwardRef>
                <ForwardRef
                  avatar={
                    <ForwardRef
                      src="https://www.randjsc.com/wp-content/uploads/2015/05/NYT-logo-square-150x150.jpg"
                    />
                  }
                  description="In the hit movie “Crazy Rich Asians,” the country looks luxe. But it’s easy to have a great time (and eat memorably well) without spending much."
                  title={
                    <ForwardRef
                      href="https://www.nytimes.com/2018/09/21/travel/singapore-travel-budget.html"
                      onClick={[Function]}
                    >
                      Singapore, for Those Who Aren’t Crazy Rich
                    </ForwardRef>
                  }
                >
                  <StyledComponent
                    avatar={
                      <ForwardRef
                        src="https://www.randjsc.com/wp-content/uploads/2015/05/NYT-logo-square-150x150.jpg"
                      />
                    }
                    description="In the hit movie “Crazy Rich Asians,” the country looks luxe. But it’s easy to have a great time (and eat memorably well) without spending much."
                    forwardedClass={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": undefined,
                        "componentStyle": ComponentStyle {
                          "componentId": "styled__Meta-npycut-2",
                          "isStatic": true,
                          "lastClassName": "eNibxo",
                          "rules": Array [
                            "display:block;",
                          ],
                        },
                        "displayName": "styled__Meta",
                        "render": [Function],
                        "styledComponentId": "styled__Meta-npycut-2",
                        "target": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                        Symbol(Symbol.hasInstance): [Function],
                      }
                    }
                    forwardedRef={null}
                    title={
                      <ForwardRef
                        href="https://www.nytimes.com/2018/09/21/travel/singapore-travel-budget.html"
                        onClick={[Function]}
                      >
                        Singapore, for Those Who Aren’t Crazy Rich
                      </ForwardRef>
                    }
                  >
                    <Meta
                      avatar={
                        <ForwardRef
                          src="https://www.randjsc.com/wp-content/uploads/2015/05/NYT-logo-square-150x150.jpg"
                        />
                      }
                      className="styled__Meta-npycut-2 eNibxo"
                      description="In the hit movie “Crazy Rich Asians,” the country looks luxe. But it’s easy to have a great time (and eat memorably well) without spending much."
                      title={
                        <ForwardRef
                          href="https://www.nytimes.com/2018/09/21/travel/singapore-travel-budget.html"
                          onClick={[Function]}
                        >
                          Singapore, for Those Who Aren’t Crazy Rich
                        </ForwardRef>
                      }
                    >
                      <div
                        className="ant-list-item-meta styled__Meta-npycut-2 eNibxo"
                      >
                        <div
                          className="ant-list-item-meta-avatar"
                        >
                          <ForwardRef
                            src="https://www.randjsc.com/wp-content/uploads/2015/05/NYT-logo-square-150x150.jpg"
                          >
                            <StyledComponent
                              forwardedClass={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": undefined,
                                  "componentStyle": ComponentStyle {
                                    "componentId": "styled__Avatar-npycut-1",
                                    "isStatic": true,
                                    "lastClassName": "LuRFV",
                                    "rules": Array [
                                      "display:block;",
                                    ],
                                  },
                                  "displayName": "styled__Avatar",
                                  "render": [Function],
                                  "styledComponentId": "styled__Avatar-npycut-1",
                                  "target": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                  Symbol(Symbol.hasInstance): [Function],
                                }
                              }
                              forwardedRef={null}
                              src="https://www.randjsc.com/wp-content/uploads/2015/05/NYT-logo-square-150x150.jpg"
                            >
                              <Avatar
                                className="styled__Avatar-npycut-1 LuRFV"
                                prefixCls="ant-avatar"
                                shape="circle"
                                size="default"
                                src="https://www.randjsc.com/wp-content/uploads/2015/05/NYT-logo-square-150x150.jpg"
                              >
                                <span
                                  className="ant-avatar styled__Avatar-npycut-1 LuRFV ant-avatar-circle ant-avatar-image"
                                  style={Object {}}
                                >
                                  <img
                                    onError={[Function]}
                                    src="https://www.randjsc.com/wp-content/uploads/2015/05/NYT-logo-square-150x150.jpg"
                                  />
                                </span>
                              </Avatar>
                            </StyledComponent>
                          </ForwardRef>
                        </div>
                        <div
                          className="ant-list-item-meta-content"
                        >
                          <h4
                            className="ant-list-item-meta-title"
                          >
                            <ForwardRef
                              href="https://www.nytimes.com/2018/09/21/travel/singapore-travel-budget.html"
                              onClick={[Function]}
                            >
                              <StyledComponent
                                forwardedClass={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": undefined,
                                    "componentStyle": ComponentStyle {
                                      "componentId": "styled__Title-npycut-5",
                                      "isStatic": true,
                                      "lastClassName": "gHRcuT",
                                      "rules": Array [
                                        "font-family:'Noto Serif KR',sans-serif;font-size:16px;font-weight:bold;",
                                      ],
                                    },
                                    "displayName": "styled__Title",
                                    "render": [Function],
                                    "styledComponentId": "styled__Title-npycut-5",
                                    "target": "a",
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                                href="https://www.nytimes.com/2018/09/21/travel/singapore-travel-budget.html"
                                onClick={[Function]}
                              >
                                <a
                                  className="styled__Title-npycut-5 gHRcuT"
                                  href="https://www.nytimes.com/2018/09/21/travel/singapore-travel-budget.html"
                                  onClick={[Function]}
                                >
                                  Singapore, for Those Who Aren’t Crazy Rich
                                </a>
                              </StyledComponent>
                            </ForwardRef>
                          </h4>
                          <div
                            className="ant-list-item-meta-description"
                          >
                            In the hit movie “Crazy Rich Asians,” the country looks luxe. But it’s easy to have a great time (and eat memorably well) without spending much.
                          </div>
                        </div>
                      </div>
                    </Meta>
                  </StyledComponent>
                </ForwardRef>
                <ForwardRef>
                  <StyledComponent
                    forwardedClass={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": undefined,
                        "componentStyle": ComponentStyle {
                          "componentId": "styled__Date-npycut-3",
                          "isStatic": true,
                          "lastClassName": "hajHJw",
                          "rules": Array [
                            "font-family:'Helvetica',sans-serif;color:#ccc;font-weight:100;font-size:12px;",
                          ],
                        },
                        "displayName": "styled__Date",
                        "render": [Function],
                        "styledComponentId": "styled__Date-npycut-3",
                        "target": "time",
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <time
                      className="styled__Date-npycut-3 hajHJw"
                    >
                      7 days ago
                    </time>
                  </StyledComponent>
                </ForwardRef>
              </div>
            </div>
            <div
              className="ant-list-item-extra"
            >
              <img
                alt="logo"
                src="https://www.randjsc.com/wp-content/uploads/2015/05/NYT-logo-square-150x150.jpg"
                width={75}
              />
            </div>
          </div>
        </div>
      </Item>
    </StyledComponent>
  </ForwardRef>
</NewsItem>
`;
